fabric-content-registries-v0/src/main/java/net/fabricmc/fabric/mixin/content/registry/FarmerVillagerTaskMixin.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.content.registry;
+
+import org.jetbrains.annotations.Nullable;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.Redirect;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+import net.minecraft.block.BlockState;
+import net.minecraft.entity.ai.brain.task.FarmerVillagerTask;
+import net.minecraft.entity.passive.VillagerEntity;
+import net.minecraft.inventory.SimpleInventory;
+import net.minecraft.item.ItemStack;
+import net.minecraft.server.world.ServerWorld;
+import net.minecraft.sound.SoundCategory;
+import net.minecraft.util.math.BlockPos;
+
+import net.fabricmc.fabric.api.registry.VillagerPlantableRegistry;
+
+@Mixin(FarmerVillagerTask.class)
+public class FarmerVillagerTaskMixin {
+	@Nullable
+	@Shadow private BlockPos currentTarget;
+
+	@Inject(method = "keepRunning", at = @At(value = "INVOKE", target = "Lnet/minecraft/entity/passive/VillagerEntity;getInventory()Lnet/minecraft/inventory/SimpleInventory;"))
+	private void fabric_useRegistry(ServerWorld serverWorld, VillagerEntity villagerEntity, long l, CallbackInfo ci) {
+		SimpleInventory simpleInventory = villagerEntity.getInventory();
+
+		for (int i = 0; i < simpleInventory.size(); ++i) {
+			ItemStack itemStack = simpleInventory.getStack(i);
+
+			if (VillagerPlantableRegistry.INSTANCE.contains(itemStack.getItem())) {
```

Comment:
Should this check for an empty stack?
What happens if a mod mistakenly registers "air" as a plantable?
This happens more often than you think if people aren't careful because "air" is both the default item and the empty marker.

Reply:
Good catch, I have added an !stack.isEmpty check